.github/workflows/main-build.yml

```diff
@@ -110,6 +117,159 @@ jobs:
           COMMIT_HASH: ${{ github.sha }}
       - uses: codecov/codecov-action@v3
 
+      - name: Get ADOT_JAVA_AGENT Image Tag
+        id: java_agent_versioning
+        working-directory: aws-otel-java-instrumentation
+        run: |
+          shortsha="$(git rev-parse --short HEAD)"
+          java_agent_tag=${{ env.ADOT_JAVA_VERSION }}-$shortsha
+          echo "STAGING_TAG=java_agent_tag" >> $GITHUB_OUTPUT
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v2.1.0
+        with:
+          role-to-assume: ${{ secrets.JAVA_INSTRUMENTATION_SNAPSHOT_ECR }}
+          aws-region: us-west-2
+
+      - name: Login to private staging ecr
+        uses: docker/login-action@v2
+        with:
+          registry: ${{ env.STAGING_ECR }}
+        env:
+          AWS_REGION: us-west-2
+
+      - name: Build and push staging image for e2e integration test
+        uses: docker/build-push-action@v4
+        with:
+          push: true
+          build-args: "ADOT_JAVA_VERSION=${{ github.event.inputs.version }}"
+          context: .
+          platforms: linux/amd64,linux/arm64
+          tags: |
+            ${{ env.STAGING_ECR }}:${{ steps.java_agent_versioning.outputs.STAGING_TAG }}
+
+  # job to build testbatches for e2e integration test
+  get-testing-suites:
+    runs-on: ubuntu-latest
+    needs: build
+    outputs:
+      test-case-batch-key: ${{ steps.set-batches.outputs.batch-keys }}
+      test-case-batch-value: ${{ steps.set-batches.outputs.batch-values }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          repository: ${{ env.TESTING_FRAMEWORK_REPO }}
+          path: testing-framework
+          ref: terraform
+
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          path: aws-otel-java-instrumentation
+
+      - name: Set up Go 1.x
+        uses: actions/setup-go@v3
+        with:
+          go-version: '~1.18.9'
+
+      - name: see env
+        run: env
+
+      - name: Create test batch key values
+        id: set-batches
+        run: |
+          cd testing-framework/tools/batchTestGenerator
+          go build
+          ./batchTestGenerator github --testCaseFilePath=./testcases.json --maxBatch=${{ env.NUM_BATCHES }} \
```

Comment:
see my comment in the other pr. I think we should host the testcases file in this repository and have a file containing only the tests that will be executed here.

Reply:
Updated accordingly.